client/request_test.go

```diff
@@ -0,0 +1,72 @@
+package client
+
+import (
+	"bytes"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"strings"
+	"testing"
+
+	"golang.org/x/net/context"
+)
+
+// TestSetHostHeader should set fake host for local communications, set real host
+// for normal communications.
+func TestSetHostHeader(t *testing.T) {
+	testURL := "/test"
+	testCases := []struct {
+		host            string
+		expectedHost    string
+		expectedURLHost string
+	}{
+		{
+			"unix:///var/run/docker.sock",
+			"",
+			"docker",
+		},
+		{
+			"npipe:////./pipe/docker_engine",
+			"",
+			"docker",
+		},
+		{
+			"tcp://0.0.0.0:4243",
+			"",
```

Comment:
Something feels wrong here.  According to this test the HTTP `Host` header will always be empty, which doesn't seem right to me.  If it really is supposed to be blank then why include the string in the test data, just check for a static `""` in the if-statement. However, having an empty header is not correct and we should fix that.

I'll try to find some time to look at this but kind of busy today....